File: 题目/题目/8.判断是否为等差数列.py
from typing import List


def canMakeArithmeticProgression(arr: List[int]) -> bool:
    arr = sorted(arr)
    t = arr[1] - arr[0]
    for i in range(1 , len(arr) - 1):
        if arr[i+1] - arr[i] != t:
            return False
    else:
        return True
        

File: 题目/题目/8.判断是否为等差数列.c
#include<stdio.h>
#include<stdlib.h>
#include<stdbool.h>

int compare(const void *arg1 , const void *arg2)
{
    return (*(int*)arg1 - *(int*)arg2);
}

bool canMakeArithmeticProgression(int* arr, int arrSize) {
    qsort(arr , arrSize , sizeof(int) , compare);
    int n = arr[1] - arr[0];
    for(int i = 1 ; i < arrSize-1 ; i++)
    {
        if (arr[i+1] - arr[i] != n)
        {
            return false;
        }
    }
    return true;
}